src/main/java/net/finmath/time/TimeDiscretization.java

```diff
@@ -29,71 +30,99 @@
 public class TimeDiscretization implements Serializable, TimeDiscretizationInterface {
 
 	private static final long serialVersionUID = 6880668325019167781L;
-	private static final double	timeTickSizeDefault = Double.parseDouble(System.getProperty("net.finmath.functions.TimeDiscretization.timeTickSize", new Double(1.0 / (365.0 * 24.0)).toString()));
+    private static final double timeTickSizeDefault = Double.parseDouble(System.getProperty("net.finmath.functions.TimeDiscretization.timeTickSize", Double.toString(1.0 / (365.0 * 24.0))));
 
 	private final double[]	timeDiscretization;
-	private final double	timeTickSize = timeTickSizeDefault;
+    private double timeTickSize;
 
 	public enum ShortPeriodLocation {
 		SHORT_PERIOD_AT_START,
 		SHORT_PERIOD_AT_END
 	}
 
+    /**
+     * Constructs a time discretization using the given tick size.
+     *
+     * @param times    A non closed and not necessarily sorted stream containing the time points.
+     * @param tickSize A non-negative double representing the smallest time span distinguishable.
+     */
+    public TimeDiscretization(DoubleStream times, double tickSize) {
+        this.timeTickSize = tickSize;
+        this.timeDiscretization = times.map(this::roundToTimeTickSize).distinct().sorted().toArray();
```

Comment:
@cfries This line takes care of the sorting. The nice thing is that all other constructors will eventually end in this constructor.

Reply:
Beautiful! I will have look. (The travis fail was my fault).